app/src/main/java/com/sdpteam/connectout/event/viewer/EventActivity.java

```diff
@@ -153,13 +159,37 @@ private void updateParticipantsButton(Event event, Button participantsBtn) {
      * (made it to avoid code duplication)
      *
      * @param fromContext from where we are starting the intent
-     * @param eventId     event Id to open with.
+     * @param eventId   event Id to open with.
      */
     public static void openEvent(Context fromContext, String eventId) {
         Intent intent = new Intent(fromContext, EventActivity.class);
         intent.putExtra(PASSED_ID_KEY, eventId);
         fromContext.startActivity(intent);
     }
 
+    /**
+     * Before joining an event, the profile must meet the registration criteria
+     * */
+    public Event.EventRestrictions.RestrictionStatus isRegistrationPossible(Profile p, Event e){
+        if(p == null){ return Event.EventRestrictions.RestrictionStatus.ALL_RESTRICTIONS_SATISFIED;} // for the null user
+        if (p.getRating() < e.getRestrictions().getMinRating()){
+            return Event.EventRestrictions.RestrictionStatus.INSUFFICIENT_RATING;
+        }
+        if (e.getParticipants().size() >= e.getRestrictions().getMaxNumParticipants()){
+            return Event.EventRestrictions.RestrictionStatus.MAX_NUM_PARTICIPANTS_REACHED;
+        }
+        if (e.getRestrictions().getJoiningDeadline() < Calendar.getInstance(TimeZone.getTimeZone("GMT+1:00")).getTimeInMillis()){
+            return Event.EventRestrictions.RestrictionStatus.JOINING_DEADLINE_PASSED;
+        }
+        return Event.EventRestrictions.RestrictionStatus.ALL_RESTRICTIONS_SATISFIED;
+    }
+
```

Comment:
I would implement the restrictions checking in a slightly different way... Following the Tell-Dont-Ask principle, I would have a class for each restriction, that can do the validation following his own logic, instead of asking (with the getters) and check them with these kinds of if.
But that's a whole other design, maybe for this small case we can keep it like this.

Reply:
interesting idea